lib/ardb.rb

```diff
@@ -91,13 +88,31 @@ def schema_path
       File.expand_path(@schema_path.to_s, @root_path.to_s)
     end
 
-    def db_settings
-      self.db.to_hash.inject({}) do |settings, (k, v)|
-        settings[k.to_s] = v if !v.nil?
-        settings
+    def schema_format=(new_value)
+      @schema_format = begin
+        new_value.to_sym
+      rescue NoMethodError
+        raise ArgumentError, "schema format must be a `Symbol`", caller
       end
     end
 
+    def activerecord_connect_hash
+      ACTIVERECORD_ATTRS.inject({}) do |h, attr_name|
+        value = self.send(attr_name)
+        !value.nil? ? h.merge!(attr_name.to_s => value) : h
```

Comment:
@kellyredding - Yeah, I used `merge!` because I figure it should be more efficient because its less objects being initialized but I'm not sure. I don't care much either way though.